### ISSUE TITLE
Interpolate from grid.

### ISSUE THREAD

**Issue by pfebrer** [zerothi/sisl#590]
I would like to add a method to interpolate values from a `Grid`, given some coordinates. I have looked into it and it seems like the way to go is to use [`scipy.interpolate.RegularGridInterpolator`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.RegularGridInterpolator.html#scipy.interpolate.RegularGridInterpolator).

To clarify, the method would work something like this:

```python

rho = sisl.Grid(...)

# Get the value of the electronic density at the points that we are interested, even
# if they are not part of the grid.
interesting_points = [[0,1.3, 4.5], [2.3, 4.7, 10]]
rho_vals = rho.interpolate(interesting_points) 
```

(1) Do you think it makes sense?
(2) Where would you put it, given that there is already a method named `interp`?

**Comment by zerothi** [comment]
I agree this would be nice.

What about allowing somekind of grid directly?
That grid could be `GridPoints` for subsets, or something similar. Much along the line of your proposal #502 ?

**Comment by pfebrer** [comment]
> What about allowing somekind of grid directly?

Do you mean interpolating the grid into another grid?

**Comment by zerothi** [comment]
yes, basically the `grid.interp` does this by just changing the shape (retaining everything else).

**Comment by pfebrer** [comment]
Ok, that is my final goal (I'm thinking of ways to speed up the computation of the psi values). But I'm not sure if there is a faster/better way than computing the coordinates of all points of your target grid and then passing them to the interpolator "individually".

Maybe with fourier transforms (?)

**Comment by zerothi** [comment]
> Ok, that is my final goal (I'm thinking of ways to speed up the computation of the psi values). But I'm not sure if there is a faster/better way than computing the coordinates of all points of your target grid and then passing them to the interpolator "individually".
> 
> Maybe with fourier transforms (?)

interpolations _can_ be quite slow unless highly optimized. So far from all times will it be good. Secondly, FFT might not always work, so backend selection should be vital.

**Comment by zerothi** [comment]
could we change `interp` to accept a `lattice` object and rescale everything there?
Or perhaps the first argument could just be another `Grid` so the lattice can be re-used, and the grid-shape taken from there.

**Comment by pfebrer** [comment]
Yes I think that there should be a base class as abstract as `Space` and then we could implement interpolation algorithms between the different subclasses of space. E.g. from `Grid` to `SpacePoints`.

Then `interp` would receive the space object into which the data should be interpolated.
